finetune/adapter.py

```diff
@@ -48,18 +47,31 @@
 }
 
 
-def main(
+def setup(
     data_dir: Path = Path("data/alpaca"),
     checkpoint_dir: Path = Path("checkpoints/stabilityai/stablelm-base-alpha-3b"),
     out_dir: Path = Path("out/adapter/alpaca"),
     precision: str = "bf16-true",
+    tpu: bool = False,
 ):
-    check_valid_checkpoint_dir(checkpoint_dir)
-
-    fabric = L.Fabric(
-        devices=devices, strategy=(DeepSpeedStrategy(config=ds_config) if devices > 1 else "auto"), precision=precision
+    strategy = (
+        "auto"
+        if devices <= 1
+        else XLAStrategy(sync_module_states=False) if tpu else DeepSpeedStrategy(config=ds_config)
     )
-    fabric.launch()
+    # For multi-host TPU training, the device count for Fabric is limited to the count on a single host.
+    fabric_devices = "auto" if (tpu and devices > 1) else devices
+    fabric = L.Fabric(devices=fabric_devices, strategy=strategy, precision="32-true" if tpu else precision)
```

Comment:
16-mixed would be a valid argument too for XLA.

I would suggest that we do instead

```
precision: Optional[str] = None


if precision is None:
    precision = "32-true" if tpu else "16-true"
```

So that Fabric can raise an error for any other value instead of silently choosing 32-true

Reply:
Thanks for the suggestion, just pushed this change.